src/zx_graph.jl

```diff
@@ -213,11 +261,29 @@ end
 Return `true` if `v` is a interior spider of `zxg`.
 """
 function is_interior(zxg::ZXGraph{T, P}, v::T) where {T, P}
-    if v in spiders(zxg)
-        nb_st = [spider_type(zxg, u) for u in neighbors(zxg, v)]
-        if !(SpiderType.In in nb_st || SpiderType.Out in nb_st)
-            return true
+    if has_vertex(zxg.mg, v)
+        (spider_type(zxg, v) == SpiderType.In || spider_type(zxg, v) == SpiderType.Out) && return false
+        for u in neighbors(zxg, v)
+            if spider_type(zxg, u) == SpiderType.In || spider_type(zxg, u) == SpiderType.Out
+                return false
+            end
         end
+        return true
     end
     return false
 end
+
+get_outputs(zxg::ZXGraph) = get_outputs(zxg.master)
+get_inputs(zxg::ZXGraph) = get_inputs(zxg.master)
+function spider_sequence(zxg::ZXGraph{T, P}) where {T, P}
+    nbits = nqubits(zxg)
+    if nbits > 0
+        vs = spiders(zxg)
+        spider_seq = Vector{Vector{T}}(undef, nbits)
+        for q = 1:nbits
+            spider_seq[q] = vs[[ZXCalculus.qubit_loc(zxg, v) == q for v in vs]]
```

Comment:
RHS will allocates, I'm wondering if we could just write this as a loop that inserts the value to `spider_seq[q]` list directly